### PR TITLE
Add agent authentication overview to agent reference

### LINKED ISSUE #2993: Describe authentication in Agent reference
### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages

observability-pipeline/observe-schedule/agent

### Expected behavior

Agent reference should provide a high-level description of how the agent authenticates to the backend via websocket transport.

### Current behavior

The term "authentication" is only used in reference to the cert-file and key-file parameters used for mTLS. Authentication via headers, the default authentication mechanism, is not described.

### Additional context

There are two supported mechanisms for agent authentication: username and password authentication, and mutual TLS authentication. I believe the relevant agent source code is here: https://github.com/sensu/sensu-go/blob/v6.2.5/agent/agent.go#L204-L220.

The following is my understanding as modeled on the source linked above:

Username and password is the default authentication mechanism. Using this mechanism sensu-agent authenticates requests to the backend by setting HTTP headers. Specifically, the agent username and password are joined by a colon and base64 encoded. The resulting string is provided as the value of `Authorization` header, e.g. `Authorization: Basic YWdlbnQ6UEBzc3cwcmQh`. When using this mechanism, the agent also uses the `Sensu-User` header to provide the username in plaintext.

Mutual TLS authentication (mTLS) is used when both the sensu-backend and sensu-agent are configured to do so. When sensu-backend is configured for mTLS, it will no longer accept username and password for authentication of agents. When sensu-agent is configured to use mTLS, it will not be able to authenticate with sensu-backend unless the backend is configured for mTLS. When configured for mTLS authentication, the agent no longer sends authentication headers.

In either authentication method, sensu-agent includes the following HTTP headers in requests sent to the backend via websocket transport:

* `Sensu-AgentName` - the agent's configured `name` as plaintext
* `Sensu-Subscriptions` - the agent's subscriptions as a comma-separated, plaintext list
* `Sensu-Namespace` - the agent's configured `namespace` as plaintext

Configuring sensu-backend for mTLS authentication requires valid certificate and key files be specified as values of `agent-auth-cert-file` and `agent-auth-key-file` parameters. Conversely, sensu-agent must be configured with valid certificate and key files as values for `cert-file` and `key-file` parameters. Both backend and agent will compare the certificate provided by the other with trusted CA certificates, either in the system trust store or specified explicitly as trusted CA via configuration.

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds agent authentication overview toward the beginning of the agent reference to cover username/password authentication and mTLS authentication mechanisms.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2993

### CODE REVIEW

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md [hunk 1]
```diff
@@ -22,11 +22,53 @@ Agent entities are responsible for creating [check and metrics events][7] to sen
 The Sensu agent is available for Linux, macOS, and Windows.
 See the [installation guide][1] to install the agent.
 
+## Agent authentication
+
+The Sensu agent authenticates to the Sensu backend via [WebSocket][45] transport by either username and password or mutual transport layer security (mTLS) authentication.
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md]:
```suggestion
The Sensu agent authenticates to the Sensu backend via [WebSocket][45] transport by either built-in basic (username and password) or mutual transport layer security (mTLS) authentication.
```

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md [hunk 2]
```diff
@@ -1803,3 +1845,4 @@ You can then use `HTTP_PROXY` and `HTTPS_PROXY` to add dynamic runtime assets, r
 [55]: ../../../commercial/
 [56]: #allow-list
 [57]: ../../observe-filter/filters#reduce-alert-fatigue-for-keepalive-events
+[58]: ../../../operations/deploy-sensu/secure-sensu/#sensu-agent-mtls-authentication
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md:1848]:
```suggestion
[58]: ../../../operations/deploy-sensu/secure-sensu/#sensu-agent-mtls-authentication
[59]: ../../../operations/control-access/#use-built-in-basic-authentication
```

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md [hunk 3]
```diff
@@ -22,11 +22,53 @@ Agent entities are responsible for creating [check and metrics events][7] to sen
 The Sensu agent is available for Linux, macOS, and Windows.
 See the [installation guide][1] to install the agent.
 
+## Agent authentication
+
+The Sensu agent authenticates to the Sensu backend via [WebSocket][45] transport by either username and password or mutual transport layer security (mTLS) authentication.
+
+### Username and password authentication
+
+Username and password authentication is the default mechanism for agent authentication.
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md]:
```suggestion
The default mechanism for agent authentication is [built-in basic authentication][59] with username and password.
```

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md [hunk 4]
```diff
@@ -22,11 +22,53 @@ Agent entities are responsible for creating [check and metrics events][7] to sen
...
 See the [installation guide][1] to install the agent.
 
+## Agent authentication
+
+The Sensu agent authenticates to the Sensu backend via [WebSocket][45] transport by either built-in basic (username and password) or mutual transport layer security (mTLS) authentication.
+
+### Username and password authentication
+
+The default mechanism for agent authentication is [built-in basic authentication][59] with username and password.
+The Sensu agent uses username and password authentication unless mTLS authentication is configured for both sensu-agent and sensu-backend.
```
Comment (cwjohnston) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md]:
```suggestion
The Sensu agent uses username and password authentication unless mTLS authentication has been explicitly configured.
```

**Review by cwjohnston — APPROVED** [review]
LGTM. I suggest getting a review from engineering in case I have provided inaccurate information in the original issue.

**Review by amdprophet — APPROVED** [review]

**Pull request merged**